docs/payloads.md

```diff
@@ -40,17 +40,17 @@ typedef struct  {
 } DocumentPayload;
 ```
 
-If no payload was set to the document, it is simply NULL. If it is not, you can go ahead and decode it.
-It is recommended to encode some metadata about the payload inside it, like a leading version number, etc.
+If no payload was set to the document, it is simply NULL. If it is not NULL, you can go ahead and decode it.
+We recommend that you encode some metadata about the payload inside it, such as a leading version number, etc.
 
-## Retrieving payloads from documents
+## Retrieving Payloads from Documents
 
 When searching, it is possible to request the document payloads from the engine. 
 
 This is done by adding the keyword `WITHPAYLOADS` to `FT.SEARCH`. 
 
-If `WITHPAYLOADS` is set, the payloads follow the document id in the returned result. 
-If `WITHSCORES` is set as well, the payloads follow the scores. e.g.:
+If `WITHPAYLOADS` is set, the payloads follow the document ID in the returned result. 
```

Comment:
What's a typical use case where we would want to fetch the payload from a document? e.g. an HTML snippet? 

I don't see the benefit of storing the data as a payload vs using a non indexed field

Reply:
The benefit is 1-2 orders of magnitude faster access to the payload than storing it in the document, plus not being forced to save the entire document. Document data needs to be fetched *for every result** from the redis main dictionary. 

 1-2 microseconds vs. ~20 nanosecond access time for the payload, over a tight loop of say 100k documents, amounts to a LOT of time.

Reply:
A typical use case would be a feature vector stored in the index vs. a feature vector extracted by the query. The score would be vector distance calculated in conjunction to the query.

Reply:
> 1-2 microseconds vs. ~20 nanosecond access time for the payload, over a tight loop of say 100k documents, amounts to a LOT of time.

I see, it's indeed a lot of time, a 1000x order of magnitude.

I don't know if I've understood correctly, from what you say, I understand that when indexing my documents, I could use the payload to store the document data as payload e.g. in JSON format `{ 'f' => 'foo', 'b' => 'bar' }`, instead of using the index as datastore, i.e. `FT.ADD {index} {docId} {score} NOSAVE`

and then, instead of fetching the fields indexed, I could fetch only the data in the payload using:

 `FT.SEARCH {index} {query} NOCONTENT WITHPAYLOADS `

And retrieving this payload will be much faster that retrieving the doc fields with `HGETALL`, which makes sense. 

If I'm right on this, then that's a game changer!

Reply:
@vruizext  You can certainly do that, but then you won't be able to return some of the fields only, or to apply highlighting (a feature that will be introduced in a couple of days). And it won't be much faster in terms of fetching the documents since we fetch only the top N documents of the result.